package-lock.json

```diff
@@ -29,6 +29,35 @@
         "wait-on": "^5.3.0"
       }
     },
+    "../mcode-extraction-framework": {
+      "version": "1.0.1",
+      "extraneous": true,
+      "license": "Apache-2.0",
+      "dependencies": {
+        "ajv": "^6.12.6",
+        "antlr4": "4.8.0",
+        "commander": "^6.2.0",
+        "csv-parse": "^4.8.8",
+        "fhir-crud-client": "^1.2.2",
+        "fhirpath": "2.1.5",
+        "lodash": "^4.17.21",
+        "moment": "^2.26.0",
+        "nodemailer": "^6.4.16",
+        "sha.js": "^2.4.9",
+        "winston": "^3.2.1"
+      },
+      "devDependencies": {
+        "@types/jest": "^26.0.14",
+        "cross-env": "^7.0.0",
+        "eslint": "^6.6.0",
+        "eslint-config-airbnb-base": "^14.0.0",
+        "eslint-plugin-import": "^2.18.2",
+        "jest": "^26.6.3",
+        "jest-when": "^2.7.0",
+        "nock": "^11.7.0",
+        "rewire": "^5.0.0"
+      }
+    },
```

Comment:
This relative path feels odd to have in a package-lock.json file. It doesn't look you have any relative paths in the `package.json` so I don't know why this got added in the branch. Do you think you can try re-running an `npm install` and see if this gets removed?

Reply:
Running npm install didn't change it. I think this might be from when I was testing the code I changed to fix the debug error, since I changed the path of the framework in package.json to point to the local copy of the framework. I changed the package.json file back after I finished testing, so I'm not sure why the package-lock.json file hasn't updated. Since it's auto-generated, can I just delete package-lock.json and run npm install again?

Reply:
You're right that it is autogenerated, so you could delete it and it would regenerate it. I think before you try that approach, I would remove the `node_modules` folder and try running `npm install` again. Sometimes if the local link is still linked in your `node_modules`, `package-lock.json` doesn't update correctly. I don't know the real reason for this, but I think it has something to do with the local linking. So try deleting `node_modules` from the root directory and then run `npm install` and it will hopefully update `package-lock.json`.

Reply:
I tried deleting node_modules and running npm install, then deleted package-lock.json and ran npm install. That fixed the problem.

Reply:
So deleting `node_modules` and running `npm install` alone didn't fix the problem?

Reply:
No, it still had the relative path. I think that part of the package-lock.json file didn't get updated for some reason.

Reply:
Hm that's odd. When you deleted the `package-lock.json` and regenerated it with `npm install` are there a lot of changes in it? Or does it just remove that one section?

Reply:
It looks like the "resolved" and "integrity" lines of each dependency were deleted, and "license" : "MIT" was added. Additionally, I just did a search in the package.json while comparing the old and new files, and it looks like the name of the line "mcode-extraction-framework": "git+https://github.com/mcode/mcode-extraction-framework.git#develop", was somehow replaced with "generic-button": "git+https://github.com/mcode/generic-button.git#develop". I renamed the nav-button CSS class a while ago using find and replace, and I think that I accidentally clicked something while doing that and caused this error.

Reply:
The current version of package.json has the correct dependency.

Reply:
Hmmm that is really odd. So nowhere in your `package-lock.json` has the usual "mcode-extraction-framework": "git+https://github.com/mcode/mcode-extraction-framework.git#develop" line? Because if the package-lock.json file was regenerated, it's pretty likely that a lot of lines would change, so it doesn't seems surprising that diff is having trouble showing the change, but it should be _somewhere_. And I'm not sure what that generic-button is for.

Reply:
While styling the form in the file-and-date-pickers branch, I changed the name of the nav-button CSS class to generic-button, because I used react-bootstrap Buttons for the file pickers, and I wanted the same styling, so the name "nav-button" no longer made sense. I used find-and-replace, so I think that I somehow accidentally replaced text in package.json after I switched to the logger-display branch. Both package.json and package-lock.json have the correct code-extraction-framework line now.

Reply:
It looks like I misread the package-lock.json file while comparing; I was on the search tab while looking at the two versions, not the source control tab, so it showed the generic-button replacement, and I thought it was the old version. I looked at the file again and realized that while the correct file path was in one spot, the local path was at another, so I deleted node_modules and package-lock.json and ran npm install while both were deleted. This fixed the problem.

Reply:
The other very simple approach to this problem could just be doing a `git reset origin/master package-lock.json` to reset the file, then you could remove `node_modules` and run `npm install` again to make sure nothing needed to change, which it shouldn't have since this branch didn't change any dependencies. That might be cleanest solution to this.

Reply:
Is this something I should do in addition to what I already did, or just an alternative if I run into this problem again in the future?

Reply:
That's just something you can do in the future if you end up with package-lock.json changes that you know you don't want to keep. Usually, you should not have any package-lock.json changes if you haven't changed package.json. Ideally, when you don't change package.json, you can `npm install` or `npm ci` (which is installing based on package-lock directly) and the install will use what is listed in the package-lock so it grabs the correct versions of packages.

In this case, I think it's okay to have upgraded the packages you have listed here.